From 83a08beb5a21f36e54fa7d04804a62aa21218915 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Tue, 15 Dec 2020 11:59:01 +0200
Subject: [PATCH 5/8] arch: arm64: imx8qm: cl-som-imx8max: enable MIPI DSI
 panel with touchscreen

Enable Startek KD050HDFIA020-C020A LCD panel on MIPI0_DSI interface.
The panel is equipped with Goodix GT911 touchscreen over MIPI_DSI0_I2C0 bus.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 .../dts/freescale/imx8qm-cl-som-imx8max.dtsi  |  7 ++
 .../boot/dts/freescale/imx8qm-sbc-imx8max.dts | 84 +++++++++++++++++++
 2 files changed, 91 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi b/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
index 66fa71e42bf2..eb3ad4eebb06 100755
--- a/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8qm-cl-som-imx8max.dtsi
@@ -691,6 +691,13 @@
 		>;
 	};
 
+	pinctrl_mipi0_lpi2c0: mipi0_lpi2c0grp {
+		fsl,pins = <
+			IMX8QM_MIPI_DSI0_I2C0_SCL_MIPI_DSI0_I2C0_SCL		0xc6000020
+			IMX8QM_MIPI_DSI0_I2C0_SDA_MIPI_DSI0_I2C0_SDA		0xc6000020
+		>;
+	};
+
 	pinctrl_lpuart0: lpuart0grp {
 		fsl,pins = <
 			IMX8QM_UART0_RX_DMA_UART0_RX				0x06000020
diff --git a/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts b/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
index b8131e9c1fad..77e093af1962 100755
--- a/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
+++ b/arch/arm64/boot/dts/freescale/imx8qm-sbc-imx8max.dts
@@ -68,6 +68,78 @@
 	};
 };
 
+/*********** MIPI-DSI ***********/
+&i2c0_mipi0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_mipi0_lpi2c0>;
+	clock-frequency = <100000>;
+	status = "okay";
+
+
+	goodix_ts@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		interrupt-parent = <&lsio_gpio1>;
+		interrupts = <19 IRQ_TYPE_LEVEL_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_touchscreen>;
+		irq-gpios = <&lsio_gpio1 19 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+};
+
+&mipi0_dphy {
+	status = "okay";
+};
+
+&mipi0_dsi_host {
+	status = "okay";
+	fsl,clock-drop-level = <2>;
+
+	panel@0 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		compatible = "startek,kd050hdfia020";
+		reg = <0>;
+		reset-gpio = <&pca9555_sb 12 GPIO_ACTIVE_HIGH>;
+		dsi-lanes = <2>;
+		status = "okay";
+
+		port@0 {
+			reg = <0>;
+			panel0_in: endpoint {
+				remote-endpoint = <&mipi0_panel_out>;
+			};
+		};
+	};
+
+	ports {
+
+		port@1 {
+			reg = <1>;
+			mipi0_panel_out: endpoint {
+				remote-endpoint = <&panel0_in>;
+			};
+		};
+	};
+};
+
+&lsio_gpio1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lsio_gpio1>;
+	status = "okay";
+
+	gpio_backlight_mipi0_dsi {
+		gpio-hog;
+		gpios = <18 GPIO_ACTIVE_HIGH>;
+		output-high;
+		line-name = "backlight-mipi0-dsi";
+	};
+};
+
 &usdhc2 { /* SD-Card*/
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
@@ -91,10 +163,22 @@
 };
 
 &iomuxc {
+	pinctrl_lsio_gpio1: lsiogpio1grp {
+		fsl,pins = <
+			IMX8QM_MIPI_DSI0_GPIO0_00_LSIO_GPIO1_IO18		0x06000020 /* MIPI_DSI0_PWM */
+		>;
+	};
+
 	pinctrl_lsio_gpio2: lsiogpio2grp {
 		fsl,pins = <
 			IMX8QM_ESAI1_TX5_RX0_LSIO_GPIO2_IO13			0x06000060 /* USB_OTG1_MUX_SEL */
 		>;
 	};
+
+	pinctrl_touchscreen: touchscreengrp {
+		fsl,pins = <
+			IMX8QM_MIPI_DSI0_GPIO0_01_LSIO_GPIO1_IO19		0x06000020 /* MIPI_DSI0_INT */
+		>;
+	};
 };
 
-- 
2.20.1

